lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -1910,6 +1925,21 @@ bool LLVMToSPIRV::isKnownIntrinsic(Intrinsic::ID Id) {
   }
 }
 
+SPIRVInstruction *LLVMToSPIRV::applyRoundingModeConstrain(Value *V,
+                                                          SPIRVInstruction *I) {
+  StringRef RMode =
+      cast<MDString>(cast<MetadataAsValue>(V)->getMetadata())->getString();
+  if (RMode == "round.tonearest")
```

Comment:
Since all rounding modes strings starts with "round.", I suggest to use `RMode.endswith("tonearest")` to minimize amount of char comparisons needed here